[CID 16689] Remove `MCObject::getname_cstring()`

peter-b opened the pull request:
Coverity Scan detected that every call to `MCObject::getname_cstring()` was creating a new C string buffer and leaking it.  This patch ensures that:

- to the greatest extent possible, `MCObject::_name` is now accessed only via `MCObject::getname()` and `MCObject::setname()`
- `MCObject::_name` has a managed lifetime by making it an `MCNewAutoNameRef`
- `MCObject::getname_cstring()` is completely removed

peter-b committed:
Access MCObject::_name only via setname() and getname()

Ensure `MCObject::_name` lifetime is correct by making sure that all
accesses are via the `setname()` and `getname()` methods.

peter-b committed:
Use managed lifetime for MCObject::_name

Ensure that the lifetime of `MCObject::_name` is managed automatically
by making it an `MCNewAutoNameRef`, avoiding the need for explicit
deletion.

peter-b committed:
[CID 16689] Remove MCObject::getname_cstring().

This method always leaks memory on use.  Code that uses
`MCObject::getname_cstring()` always assume the pointer it returns is
owned by the `MCObject` instance, when in fact the method returns a
newly allocated string buffer.

Convert call sites to use an `MCAutoStringRefAsCString` instead.

Coverity-ID: 16689

peter-b committed:
Remove pre-XP support for non-Unicode Windows window creation.

runrevmark commented:
@livecode-vulcan review ok 8204aa8

livecode-vulcan commented:
💙 review by @runrevmark ok 8204aa89f525d189e6b5516bc05abfdbc204ca96

livecode-vulcan commented:
😞 test failure 8204aa89f525d189e6b5516bc05abfdbc204ca96

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/739)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/744)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/737)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/738)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/744)
* try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/735)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/736)

runrevmark commented:
@livecode-vulcan review ok e6f55df

livecode-vulcan commented:
💙 review by @runrevmark ok e6f55dfdc4ddaf40b75ad4c82572dbc4858bb00b

livecode-vulcan commented:
😞 test failure e6f55dfdc4ddaf40b75ad4c82572dbc4858bb00b

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/740)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/745)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/738)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/739)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/745)
* try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/736)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/737)

peter-b commented:
@runrevmark I've had to poke this again because it turns out that you can't lock an `MCAutoStringRefAsWString` with an `MCNameRef`. :-(

peter-b committed:
[CID 16689] Fix possible over-release in test mode engine.

The string returned by `MCModeHandleRelaunch()` is eventually released
by the caller.

livecodeali commented on engine/src/mode_test.cpp:
```diff
@@ -490,15 +494,13 @@ bool MCModeHandleMessageBoxChanged(MCExecContext& ctxt, MCStringRef)
 	t_do_relaunch = MCdefaultstackptr -> hashandler(HT_MESSAGE, MCM_relaunch) == True;
-	t_id = MCdefaultstackptr -> getname_cstring();
-	r_id = t_id;
+	r_id = MCNameGetString(getname());
```
Callers of `MCModeHandleRelaunch` appear to own the id string, so I think this needs a retain.

livecodeali replied:
I say callers, there's only one

livecodeali commented:
@livecode-vulcan review ok b9b12a8

livecode-vulcan commented:
💙 review by @livecodeali ok b9b12a8609022035d1580f166519ba7579a79eaf

livecode-vulcan commented:
😎 test success b9b12a8609022035d1580f166519ba7579a79eaf

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/751)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/756)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/749)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/750)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/756)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/748)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/748)
